task/sast-coverity-buildless-check/0.1/sast-coverity-buildless-check.yaml

```diff
@@ -0,0 +1,193 @@
+    # TODO: Change image
+      image: quay.io/redhat-user-workloads/sast-tenant/sast-scanner/coverity@sha256:ce98d4a80a2b77cd1b51fb11d563515226331fdf9521daa1b82da5b0a99e8d22
+      computeResources:
+        requests:
+          memory: "16Gi"
+          cpu: "8"
+        limits:
+          memory: "32Gi"
+          cpu: "16"
+      # per https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting
+      # the cluster will set imagePullPolicy to IfNotPresent
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      volumeMounts:
+        - name: cov-license
+          mountPath: "/etc/secrets"
+          readOnly: true
+      env:
+        - name: COV_ANALYZE_ARGS
+          value: $(params.COV_ANALYZE_ARGS)
+        - name: COV_CAPTURE_ARGS
+          value: $(params.COV_CAPTURE_ARGS)
+        - name: KFP_GIT_URL
+          value: $(params.KFP_GIT_URL)
+        - name: SEVERITY_THRESHOLD
+          value: $(params.SEVERITY_THRESHOLD)
+      script: |
+        #!/usr/bin/env bash
+        set -eo pipefail
+        . /usr/local/share/konflux-test/utils.sh
+        trap 'handle_error $(results.TEST_OUTPUT.path)' EXIT
+
+        echo 'Starting Coverity buildless scan'
+        SOURCE_CODE_DIR=$(workspaces.workspace.path)
+        COVERITY_DIR=/tmp/cov-scan/cov
+        COVERITY_RESULTS_FILE=coverity-buildless-results.js
+        COV_LICENSE_PATH=/etc/secrets/cov-license
+        # Installing Coverity license
+        cp /etc/secrets/cov-license /opt/coverity/bin/license.dat
+        if test -z "$(find /opt -name 'license.dat')"; then
+          echo "No license file for Coverity was detected. Exiting..."
+          exit 1
+        fi
+        
+        # Installation of Red Hat certificates for cloning Red Hat internal repositories
+        curl -sS https://certs.corp.redhat.com/certs/2015-IT-Root-CA.pem > /etc/pki/ca-trust/source/anchors/2015-RH-IT-Root-CA.crt
+        curl -sS https://certs.corp.redhat.com/certs/2022-IT-Root-CA.pem > /etc/pki/ca-trust/source/anchors/2022-IT-Root-CA.pem
+        update-ca-trust
+        
+        # Create configuration file for coverity buildless
+        echo -e 'capture:\n   build-command-inference: false' > $SOURCE_CODE_DIR/coverity.yml
+        
+        # Captureless scan
+        env COV_HOST=konflux HOME=/tmp/cov-scan/home /opt/coverity/bin/coverity capture $COV_CAPTURE_ARGS --project-dir $SOURCE_CODE_DIR --dir $COVERITY_DIR
+        COV_CAPTURE_EXIT_CODE=$?
+        
+        if [[ "$COV_CAPTURE_EXIT_CODE" -eq 0 ]]; then
+            echo "Coverity capture scan finished succesfully"
+        else
+            echo "Coverity capture command failed. Exiting..."
+            exit 1
+        fi
+        
+        # Analysis phase
+        /opt/coverity/bin/cov-manage-emit --dir $COVERITY_DIR reset-host-name
+        /opt/coverity/bin/cov-analyze $COV_ANALYZE_ARGS --dir=$COVERITY_DIR --enable HARDCODED_CREDENTIALS --security --concurrency --spotbugs-max-mem=4096
```

Comment:
I would move `--security --concurrency --spotbugs-max-mem=4096` to the default value of `COV_ANALYZE_ARGS`.  Otherwise users would have no means to override these options.

Reply:
And not the `--enable HARDCODED_CREDENTIALS` àrameter ?

Reply:
Yep, that one, too.  I have shortened the copy/pasted text by mistake.

Reply:
Added